File: graph-java/src/main/java/com/cc/graph/Conf.java
package com.cc.graph;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;

public final class Conf {

    private final static Config conf = ConfigFactory.load();

    public final static String projectName = conf.getString("project.name");

    public final static class Gep {
        public static int generationNum = conf.getInt("gep.max-generation-num");
        public static int populationSize = conf.getInt("gep.population-size");
        public static double geneMove = conf.getDouble("gep.gene-move");
        public static double geneExchange = conf.getDouble("gep.gene-exchange");
        public static double geneSplitoff = conf.getDouble("gep.gene-splitoff");
        public static double geneMerge = conf.getDouble("gep.gene-merge");
    }
}

File: graph-dynamic/src/main/scala/com/cc/graph/Conf.scala
package com.cc.graph

import com.typesafe.config.ConfigFactory

/**
 * configuration instance
 */
object Conf {
  private lazy val conf = ConfigFactory.load();

  lazy val projectName = conf.getString("project.name")

  object Gep {
    lazy val generationNum = conf.getInt("gep.max-generation-num")
    lazy val populationSize = conf.getInt("gep.population-size")
    lazy val geneMove = conf.getDouble("gep.gene-move")
    lazy val geneExchange = conf.getDouble("gep.gene-exchange")
    lazy val geneSplitoff = conf.getDouble("gep.gene-splitoff")
    lazy val geneMerge = conf.getDouble("gep.gene-merge")
  }
}